Libraries/Text/RCTTextView.m

```diff
@@ -127,6 +128,20 @@ - (NSString *)text
 
 - (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text
 {
+  /**
+   * Only allow single keypresses for onKeyPress, most text pasted in will not be sent.
+   *
+   * Unfortunately, UITextView for some reason doesn't implement `paste:sender` like UITextField, so we have
+   * no way of distinguishing a single character paste from a single character keyPress. :(
+   * I'm assuming this is a bug in iOS, but I couldn't find out for sure.
+   *
+   * End result: Single character pastes will register and fire a keyPress event with the character
+   * that was pasted.
+   */
+  if (text.length <= 1) {
```

Comment:
According to http://stackoverflow.com/questions/19045484/how-to-know-when-paste-event-triggered-in-uitextview it does work, but you need to set the delegate of the TextView or it won't fire.

The delegate is currently set to itself, which may be why it isn't working. @tadeuzagallo has just changed this though because it was causing other issues, so it may work once his fix has landed.